internal/pkg/agent/application/upgrade/details/state.go

```diff
@@ -0,0 +1,22 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package details
+
+type State string
+
+// The values of these State* constants should match those enumerated for
+// upgrade_details.state in https://github.com/elastic/fleet-server/blob/main/model/openapi.yml
+const (
+	StateRequested   State = "UPG_REQUESTED"
+	StateScheduled   State = "UPG_SCHEDULED"
```

Comment:
> But this means we are conflating the Fleet datamodel for upgrade status and the internal Agent datamodel as one and the same, which I am not too happy with.

I was thinking about this and, at least at the moment, I can't think of any upgrade states that Fleet would want to know about but Agent might not or vice versa.  So I think it's OK to use the same datamodel.  In fact, to take it a step further and make the 1-1 nature even more explicit and enforced, it would be good to generate the these constants from the Fleet Open API spec as @cmacknz suggested and I've captured as a follow up task in https://github.com/elastic/elastic-agent/issues/3119#issuecomment-1743638526.

I agree with you that the moment we see the 1-1 correspondence breaking, we should separate the two data models.